CMakeLists.txt

```diff
@@ -0,0 +1,230 @@
+cmake_minimum_required(VERSION 3.10.2)
+project(TheStdandardLib
+    VERSION 1.0
+    LANGUAGES C
+)
+
+option(TARGET_X86_64
+    "Default target device is set to x86_64"
+    "ON"
+)
+
+set(CMAKE_ANDROID_ARCH_ABI "unsupported_cpu_architecture")
+if(TARGET_X86_64)
+    set(CMAKE_ANDROID_ARCH_ABI "x86_64")
+    set(ANDROID_ABI "x86_64") # FIXME: Don't know how keep compatable with Android NDK
+    set(CMAKE_ASM_NASM_COMPILE_OPTIONS "${CMAKE_ASM_NASM_COMPILE_OPTIONS} -f elf64 -g")
+endif()
+if(CMAKE_ANDROID_ARCH_ABI STREQUAL "unsupported_cpu_architecture")
+    message(WARNING "Currently, only X86_64 is fully supported. (FIXME: Other none-X86 target devices are going to be supported in the future)")
+endif()
+
+include(CheckLanguage)
+check_language(ASM_NASM)
+if(CMAKE_ASM_NASM_COMPILER)
+    enable_language(ASM_NASM)
+endif()
+if (NOT CMAKE_ASM_NASM_COMPILER_LOADED)
+    message(WARNING "CMAKE_ASM_NASM_COMPILER is not found...\n"
+        "X86_64 related features will not be available!\n"
+        "Suggest: sudo apt-get install yasm")
+endif()
+
+message(STATUS "CMAKE_ASM_NASM_COMPILER = " ${CMAKE_ASM_NASM_COMPILER})
+message(STATUS "CMAKE_ASM_NASM_COMPILER_LOADED = " ${CMAKE_ASM_NASM_COMPILER_LOADED})
+message(STATUS "CMAKE_ASM_NASM_COMPILE_OPTIONS = " ${CMAKE_ASM_NASM_COMPILE_OPTIONS})
+
+set(CMAKE_ASM_NASM_SOURCE_FILE_EXTENSIONS S)
+message(STATUS "CMAKE_ASM_NASM_SOURCE_FILE_EXTENSIONS = " ${CMAKE_ASM_NASM_SOURCE_FILE_EXTENSIONS})
+set(X86_64_ASM_NASM_SOURCES
+    tsl/coro/x86_64_coro_swap.S
+)
+if(CMAKE_ANDROID_ARCH_ABI STREQUAL "x86_64")
+    set(ASM_NASM_SOURCES ${X86_64_ASM_NASM_SOURCES})
+elseif(CMAKE_ANDROID_ARCH_ABI STREQUAL "aarch64")
```

Comment:
I think this would break all `aarch64` builds, not just prevent building some of the x86-64 specific sources. `armv7l` and `armv8l`.